readthedocs/oauth/views.py

```diff
@@ -0,0 +1,557 @@
+                 },
+                 "repository_selection": "selected",
+                 "html_url": "https://github.com/settings/installations/1234",
+                 "app_id": 12345,
+                 "app_slug": "app-slug",
+                 "target_id": 12345,
+                 "target_type": "User",
+                 "permissions": {
+                     "contents": "read",
+                     "metadata": "read",
+                     "pull_requests": "write",
+                     "statuses": "write"
+                 },
+                 "events": [
+                     "create",
+                     "delete",
+                     "public",
+                     "pull_request",
+                     "push"
+                 ],
+                 "created_at": "2025-01-29T12:04:11.000-05:00",
+                 "updated_at": "2025-01-29T12:04:12.000-05:00",
+                 "single_file_name": null,
+                 "has_multiple_single_files": false,
+                 "single_file_paths": [],
+                 "suspended_by": null,
+                 "suspended_at": null
+             },
+             "repositories": [
+                 {
+                     "id": 770738492,
+                     "name": "test-public",
+                     "full_name": "user/test-public",
+                     "private": false
+                 },
+                 {
+                     "id": 917825438,
+                     "name": "test-private",
+                     "full_name": "user/test-private",
+                     "private": true
+                 }
+             ],
+             "requester": null,
+             "sender": {
+                 "login": "user",
+                 "id": 1234,
+                 "type": "User"
+              }
+           }
+
+        See https://docs.github.com/en/webhooks/webhook-events-and-payloads#installation.
+        """
+        data = self.request.data
+        action = data["action"]
+        gh_installation = data["installation"]
+        installation_id = gh_installation["id"]
+
+        if action in ["created", "unsuspended"]:
+            installation, created = self._get_or_create_installation()
+            # If the installation was just created, we already synced the repositories.
+            if created:
+                return
+            installation.service.sync()
+
+        if action in ["deleted", "suspended"]:
+            # NOTE: When an installation is deleted/suspended, it doesn't trigger an installation_repositories event.
+            # So we need to call the delete method explicitly here, so we delete its repositories.
+            installation = GitHubAppInstallation.objects.filter(
+                installation_id=installation_id
+            ).first()
+            if installation:
+                installation.delete()
+                log.info("Installation deleted")
+            else:
+                log.info("Installation not found")
```

Comment:
These are really hard log messages to understand what to do with. We should be adding more context about if this is an issue, otherwise I'm not really sure what this log message is useful for? What will we understand when we see it in NR?

Reply:
They are useful to debug things in case we need to know why a user doesn't see their repos, they are marked as info, so there is no need to take any action.